crates/biome_js_analyze/src/lint/nursery/use_default_case.rs

```diff
@@ -0,0 +1,195 @@
+use std::ops::Not;
+
+use biome_analyze::{context::RuleContext, declare_rule, Ast, Rule, RuleDiagnostic, RuleSource};
+use biome_console::markup;
+use biome_deserialize_macros::Deserializable;
+use biome_js_syntax::{JsLanguage, JsSwitchStatement};
+use biome_rowan::{syntax::SyntaxTrivia, AstNode};
+use regex::Regex;
+use serde::{Deserialize, Serialize};
+
+declare_rule! {
+    /// Require default cases in switch statements.
+    ///
+    /// See https://eslint.org/docs/latest/rules/default-case
+    ///
+    /// ## Examples
+    ///
+    /// ### Invalid
+    ///
+    /// ```js,expect_diagnostic
+    /// switch (a) {
+    ///     case 1:
+    ///         /* code */
+    ///         break;
+    /// }
+    /// ```
+    ///
+    /// ```js,expect_diagnostic
+    /// switch (a) {
+    ///     // no default
+    /// 	case 1:
+    ///         break;
+    /// }
+    /// ```
+    ///
+    /// ```js,expect_diagnostic
+    /// switch (a) {
+    ///     case 1:
+    ///         break;
+    ///     // no default
+    ///     // nope
+    /// }
+    /// ```
+    ///
+    /// ### Valid
+    ///
+    /// ```js
+    /// switch (a) {
+    ///     case 1:
+    ///         /* code */
+    ///         break;
+    ///
+    ///     default:
+    ///         /* code */
+    ///         break;
+    /// }
+    /// ```
+    ///
+    /// ```js
+    /// switch (a) {
+    ///     case 1:
+    ///         /* code */
+    ///         break;
+    ///
+    ///     // no default
+    /// }
+    /// ```
+    ///
+    /// ```js
+    /// switch (a) {
+    /// }
+    /// ```
+    pub UseDefaultCase {
+        version: "next",
+        name: "useDefaultCase",
```

Comment:
For maintaining consistency with other rule names, I suggest renaming the rule to `useDefaultSwitchClause`.

Reply:
done